modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -310,87 +282,61 @@ def get_files_info(ip, _rport, shares, info)
     end
   end
 
-  def rport
-    @rport || datastore['RPORT']
-  end
-
-  # Overrides the one in smb.rb
-  def smb_direct
-    @smb_redirect || datastore['SMBDirect']
-  end
-
   def run_host(ip)
-    @rport = datastore['RPORT']
-    @smb_redirect = datastore['SMBDirect']
-    @srvsvc = datastore['USE_SRVSVC_ONLY']
     shares = []
 
-    [[139, false], [445, true]].each do |info|
-      @rport = info[0]
-      @smb_redirect = info[1]
+    [{ port: 139, redirect: false }, { port: 445, redirect: true} ].each do |info|
+      @rport = info[:port]
+      @smb_redirect = info[:redirect]
 
       begin
-        connect
+        print_status 'Starting module'
+        if rport == 139
+          connect(versions: [1])
+        else
+          connect(versions: [1, 2, 3])
+        end
         smb_login
-        shares = smb_netshareenumall
-
-        os_info = get_os_info(ip, rport)
+        shares = simple.client.net_share_enum_all(ip)
```

Comment:
No idea why this was marked as resolved, as its clearly not. This can end up calling https://github.com/rapid7/ruby_smb/blob/bacb5428d94121607f0c77cd31929afcd49d30a0/lib/ruby_smb/client/tree_connect.rb#L72 which will raise a `RubySMB::Error::InvalidPacket` error. There are likely other errors that could be raised here though given the complexity of RubySMB and handling request and responses though.

Reply:
Fixed in commit 1a435f8a774817d4a328d542c831ee0b63a398a0, thanks for addressing this!